resources/articles/s2s-auth.md

```diff
@@ -0,0 +1,121 @@
+
+The following sections break down these high-level steps in more detail.
+
+## Obtaining a token
+
+Before making a call to a resource server, the calling client first needs to [obtain a token](/oauth/token/).
+For this, a `client_credentials` grant is made using the caller's client ID and secret. Additionally,
+the caller should specify the resource (a.k.a. audience) trying to be accessed via the `resource`
+parameter and any scopes required via the `scope` parameter:
+
+```sh
+curl -X POST -H "Authorization: Basic NGU4NDYzNTY5Y2FmN2NhMDE5MDAwMDA3OmZvb2Jhcg" \
+-d 'resource=https://service.schibsted.com/messages' -d 'scope=read:messages write:messages' \
+-d 'grant_type=client_credentials' https://login.schibsted.com/oauth/token
+```
+
+For further details on resource indicators, please refer to
+[Resource Indicators for OAuth 2.0, draft 2](https://tools.ietf.org/html/draft-campbell-oauth-resource-indicators-02).
+
+***Note:*** currently the value passed as the resource needs to match the domain registered in SPiD's
+[self service](/mobile/selfservice/) tool for the resource server.
+
+The scopes passed will help the receiving service authorize the call, and can be thought of a set of
+permissions or policies that the service will allow. Thus, the scopes present in the token help the
+called service determine if the request actions should be permitted. The exact scopes required is
+determined by the resource server and is outside the scope of this document.
+
+**Note:** currently the scopes that a client may request may only be managed by the administrators of
+the SPiD platform. In order to have scopes added to your client [contact support](mailto:support@spid.no).
+In the future, management of scopes will be possible via the [self service](/mobile/selfservice/) tool.
+
+## Send the token in the request
+
+Next, when making the request to the resource server, the caller simply needs to include the freshly
+acquired token in the Authorization header as a [bearer token](https://tools.ietf.org/html/rfc6750):
+
+```sh
+curl -H 'Authorization: Bearer _your_token_here' https://service.schibsted.com/messages/123
+```
+
+## Validate the token
+
+The resource server receives the token in `Authorization` header and must validate it in order to decide
+whether or not to allow the call. The first step in doing so is to [introspect the token](/token-introspection/)
+in order to get accessed to the claims it carries. Introspecting the token will provide a result similar
+to the following:
+
+```json
+{
+  "scope": "read:messages",
+  "client_id": "4e8463569caf7ca019000007",
+  "token_type": "Bearer",
+  "exp": 1487169930,
+  "iat": 1487166577,
+  "iss": "https://login.schibsted.com",
+  "aud": ["https://service.schibsted.com/messages"]
+}
+```
+
+One the resource server has introspected the token, it's important to validate the claims as follows:
+1. Has the token expired? This is determined by checking the `exp` claim of the token. When performing
```

Comment:
Missing new line before this to render list correctly.

Reply:
Fixed